### PR TITLE
Automatically generate screenshots for PRs

### PR DESCRIPTION
**Pull request by patrick-stephens**
Workflow to create screenshots.
Can also be run via `make generate-screenshots` locally.
Not detaining the screenshots in repo currently, just in the action artifact.

### CODE REVIEW

**Comment by patrick-stephens** [comment]
Currently not running with data but want to get the workflow merged so we can test it.

File: testing/screenshots/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    const screenshots = [];
+
+    await Promise.all(files.map(async file => {
+        const absolute = path.resolve(__dirname, "../..", file);
+        const base = path.basename(file).replace(".json", "");
+        const definition = require(absolute);
+        const uid = definition.uid;
+
+        const page = await browser.newPage();
+        await page.setViewport({ width: 1440, height: 1080 });
```
Comment (patrick-stephens) [inline testing/screenshots/index.js:62]: I presume we should set this to what we want

File: testing/screenshots/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        await page.goto(`http://localhost:8080/grafana/d/${uid}`);
```
Comment (patrick-stephens) [inline testing/screenshots/index.js:63]: Make this configurable in next iteration too so we can just run and fetch the ephemeral port like we do with smoke tests

File: testing/screenshots/index.js (hunk 1 above)
Comment (markspolakovs) [reply testing/screenshots/index.js:62]: Yup - in theory we could also make it loop over a set of viewport sizes if we want multiple screenshots to be taken

File: .github/workflows/pr-screenshots.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+            - main
+        paths:
+            - microlith/grafana/provisioning/dashboards/**
+
+    workflow_dispatch:
+        inputs:
+            ref:
+                description: Branch, tag, or commit SHA to run against
+                required: false
+                default: main
```
Comment (markspolakovs) [inline .github/workflows/pr-screenshots.yml:16]: Looks like you're defining `$ref` but not using it anywhere?

Comment (patrick-stephens) [reply .github/workflows/pr-screenshots.yml:16]: Ah yeah, wanted to add that to checkout

**Review by markspolakovs — APPROVED** [review]
Happy with this as a first version, we can always iterate.

**Pull request merged**